src/main/java/org/springframework/integration/dsl/config/IntegrationFlowBeanPostProcessor.java

```diff
@@ -143,6 +160,7 @@ private Object processStandardIntegrationFlow(StandardIntegrationFlow flow, Stri
 					endpointBeanName = generateBeanName(endpoint);
 				}
 				registerComponent(endpoint, endpointBeanName, beanName, registerSingleton);
+				integrationComponents.set(i, endpoint);
```

Comment:
This seems a little dirty - relying on the getter returning a hard reference to the field and updating it from here.

On the other hand, it's not clear to me what this is doing - replacing the `ConsumerEndpointSpec` with the factory bean would make sense (since the factory bean is a `SmartLifecyle`) but we've already built the `lifecycles` field in the `StandardIntegrationFlow`'s constructor. So how does `stop()/start()` work? The spec it replaces is not a `Lifecycle`.